platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/internal/extensibility/DeprecatedBooleanPropertyIntegrationTest.groovy

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.extensibility
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class DeprecatedBooleanPropertyIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
Does this test run in the Gradleception build with Groovy 4? If not, can we make one that does?

Reply:
No, I'm not sure how easy it would be to add a non-integration test, and I don't think we want to add integration tests to Gradleception.

Reply:
After talking elsewhere, we determined it would be best to add `:model-core:forkingIntegTest` to Gradleception + Groovy 4 to get some coverage here.